### PR TITLE
IDAM-612 Convert Post Change Password Hook To Action

### PR DESCRIPTION
**Pull request by AlexPowell-nice**
https://nicedigital.atlassian.net/browse/IDAM-612

### CODE REVIEW

File: custom-pages/actions/Post Change Password - Create IDAM User/code.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  
+  if (tokenResponse.status !== 200) {
+    const errorMessage = `Error in token call: HTTPCode - ${tokenResponse.status} | Error - ${tokenResponseBody.error} | Description - ${tokenResponseBody.error_description}`;
+    console.log(errorMessage);
+    throw new Error(errorMessage);
+  }
+  
+  console.log("Post Change Password - Checking to see if user exists in IDAM");
+
+  var userExistsResponse;
```
Comment (tresaknice) [inline custom-pages/actions/Post Change Password - Create IDAM User/code.js:45]: Can "var" be changed to "let" to keep it consistent with the rest of the code.

File: custom-pages/actions/Post Change Password - Create IDAM User/code.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+  if (userExists){
+    console.log(`Post Change Password - Identity account for that user already exists: ${event.user.email}`);
+    return;
+  }
+  
+  //user doesn't exist. this could happen if the user has reset a password for a user that is in nice accounts
+  console.log("Post Change Password - Checking NICE Accounts to see if user exists there");
+  
+  var checkNICEAccountsResponse;
```
Comment (tresaknice) [inline custom-pages/actions/Post Change Password - Create IDAM User/code.js:80]: Can "var" be changed to "let"

File: custom-pages/actions/Post Change Password - Create IDAM User/code.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  }
+
+  if (checkNICEAccountsResponse.status !== 200){
+    const errorMessage = `Post Change Password - NICE Accounts error: HTTPCode - ${checkNICEAccountsResponse.status} | Status - ${checkNICEAccountsResponse.statusText}`;
+    console.log(errorMessage);
+    throw new Error(errorMessage);
+  }
+  
+  //at this point nice accounts has returned a valid user, which doesn't exist in the idam db, so we just need to hit our db with the user.
+  let user = JSON.parse(await checkNICEAccountsResponse.text())
```
Comment (tresaknice) [inline custom-pages/actions/Post Change Password - Create IDAM User/code.js:111]: can this be changed to const as the user is not being reassigned

File: custom-pages/actions/Post Change Password - Create IDAM User/code.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+  const postData = JSON.stringify({
+    'nameIdentifier': `auth0|${user.user_id}`,
```
Comment (tresaknice) [inline custom-pages/actions/Post Change Password - Create IDAM User/code.js:114]: would there be a situation where "user" could be null?

File: custom-pages/actions/Post Change Password - Create IDAM User/code.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    'acceptedTerms': false,
+    'allowContactMe': false,
+    'hasVerifiedEmailAddress': true,
+    'isLockedOut': false,
+    'isMigrated': true,
+    'isStaffMember':false,
+    'isInAuthenticationProvider': false //the user won't get created in auth0 db until they log in.
+  });
+  
+  var addUserResponse;
```
Comment (tresaknice) [inline custom-pages/actions/Post Change Password - Create IDAM User/code.js:127]: Can "var" be changed to "let"

**Review by tresaknice — CHANGES_REQUESTED** [review]

**Review by tresaknice — APPROVED** [review]

**Review by mgallagher1980 — DISMISSED** [review]

**Review by mgallagher1980 — APPROVED** [review]

**Pull request merged**